projects/admin/src/app/record/circulation-logs/circulation-logs.component.ts

```diff
@@ -105,13 +108,44 @@ export class CirculationLogsComponent implements OnInit {
       });
   }
 
+  /**
+   * This function will check if two date are in the same time period
+   * @param date1: the first date to check.
+   * @param date2: the second date to check.
+   * @param period: the time period to use to compare both dates
+   * @return: True if dates are similar, False otherwise
+   */
+  isSamePeriod(date1: any, date2: any, period: string = 'month'): boolean {
+    if (date1 && date2) {
+      const transDate1 = moment(date1.metadata.date);
+      const transDate2 = moment(date2.metadata.date);
+      return transDate1.isSame(transDate2, period);
+    }
+    return true;
+  }
+
+  /**
+   * Handler for mouse event
+   * @param record: the record on which the event is triggered.
+   */
+  mouseEventTransaction(record: any): void {
```

Comment:
Do you need to filter by mouse event?

Reply:
In a first implementation, I was listening about 'onmouveenter' and 'onmouseleave'.
After a small refactoring, this function is now only listening 'onmouseenter'. But maybe, a future day, this function could listen other events.